hail/src/main/scala/is/hail/types/physical/PCanonicalArray.scala

```diff
@@ -161,6 +161,8 @@ final case class PCanonicalArray(elementType: PType, required: Boolean = false)
       firstElementOffset(aoff, loadLength(aoff)) + i.toL * const(elementByteSize)
     }
 
+  private def elementOffsetFromFirst(addr: Code[Long], i: Code[Int]): Code[Long] = addr + i.toL * const(elementByteSize)
```

Comment:
Maybe just call the argument `firstElementAddr` or something? This is the only function that takes an address to the first element that I'm aware of. Also, the name sort of sounds like it might be the number of bytes between the first element and element at position `i`, but that's maybe just part of using the word offset everywhere at this point.

Reply:
good point.